Taskfile.yaml

```diff
@@ -30,29 +30,45 @@ tasks:
   lint-java:
     desc: "Run linter for java"
     sources:
-      - hive-udf/**/*
+      - hive-udf/gradle/**/*
+      - hive-udf/lib/src/**/*
+      - hive-udf/lib/build.gradle
+    generates:
+      - featurebyte/sql/spark/*.jar
     cmds:
         - cd hive-udf && ./gradlew spotlessCheck
 
   format-java:
     desc: "Format the java code"
     sources:
-      - hive-udf/**/*
+      - hive-udf/gradle/**/*
+      - hive-udf/lib/src/**/*
+      - hive-udf/lib/build.gradle
+    generates:
+      - featurebyte/sql/spark/*.jar
     cmds:
       - cd hive-udf && ./gradlew spotlessApply
 
   test-java:
     desc: "Run the java tests"
     sources:
-      - hive-udf/**/*
+      - hive-udf/gradle/**/*
+      - hive-udf/lib/src/**/*
+      - hive-udf/lib/build.gradle
+    generates:
+      - featurebyte/sql/spark/*.jar
     cmds:
       - cd hive-udf && ./gradlew test
 
   build-jar:
     desc: "Compile the hive jar files"
     run: once
     sources:
-      - hive-udf/**/*
+      - hive-udf/gradle/**/*
+      - hive-udf/lib/src/**/*
+      - hive-udf/lib/build.gradle
+    generates:
+      - featurebyte/sql/spark/*.jar
```

Comment:
Curious if there is a way in Taskfile to reuse common parameters between tasks? This seems a bit repetitive.

Reply:
we tried haha. We will try harder in another PR haha